### PR TITLE
Enable .NET 7 NativeAOT and binary trimming.

### PR DESCRIPTION
**Pull request by hjc4869**
Need CI config updates to enable .NET 7

Binary size: https://github.com/hjc4869/EnergyStar/releases/tag/0.0.1

### CODE REVIEW

**Comment by LifeIsStrange** [comment]
AOT has lower throughput and energy efficiency than JIT. Therefore it should not be the default user experience.
Can make sense for people that have extremely small disk space though but very niche.

File: EnergyStar/EnergyStar.csproj [hunk 1]
```diff
@@ -3,13 +3,30 @@
...
 
+  <!-- Optimize for size -->
+  <PropertyGroup Condition="'$(Configuration)'=='Release' or '$(Configuration)'=='ReleaseInvisible'">
+    <PublishAot>true</PublishAot>
+    <IlcOptimizationPreference>Size</IlcOptimizationPreference>
+    <IlcFoldIdenticalMethodBodies>true</IlcFoldIdenticalMethodBodies>
+    <IlcGenerateStackTraceData>false</IlcGenerateStackTraceData>
+    <TrimUnusedDependencies>true</TrimUnusedDependencies>
+    <RootPackageReference>false</RootPackageReference>
+    <TrimMode>link</TrimMode>
```
Comment (hez2010) [inline EnergyStar/EnergyStar.csproj]: `TrimMode=link` has been deprecated in .NET 7. Use `TrimMode=full`instead.

**Review by hez2010 — CHANGES_REQUESTED** [review]

**Comment by hez2010** [comment]
> AOT has lower throughput and energy efficiency than JIT.

.NET NativeAOT has the same throughput with JIT tier2 optimization, and it has lower resource consumption.
It's suitable for client apps because PGO is not that important rather than server apps.

File: EnergyStar/EnergyStar.csproj (hunk 1 above)
Comment (hjc4869) [reply EnergyStar/EnergyStar.csproj]: Updated

**Comment by hez2010** [comment]
Since this app doesn't use reflection at all, you can also add `<IlcDisableReflection>true</IlcDisableReflection>` to shink the app size to 1.6mb.

**Comment by JasonWei512** [comment]
> 



> Since this app doesn't use reflection at all, you can also add `<IlcDisableReflection>true</IlcDisableReflection>` to shink the app size to 1.6mb.

The `feafures/config` branch uses `System.Text.Json` to read json config file.

**Comment by hez2010** [comment]
> The `feafures/config` branch uses `System.Text.Json` to read json config file.

In this case you may want to use `JsonDocument` APIs and parse json directly without using `JsonSerailizer`.

**Comment by imbushuo** [comment]
Stage this PR to NativeAOT branch (waiting for .NET 7 production release...)

**Pull request merged**